cmd/clusterctl/pkg/client/cluster/inventory.go

```diff
@@ -68,6 +68,17 @@ func newInventoryClient(proxy Proxy) *inventoryClient {
 }
 
 func (p *inventoryClient) EnsureCustomResourceDefinitions() error {
+	c, err := p.proxy.NewClient()
+	if err != nil {
+		return err
+	}
+
+	// check the CRD already exists, if yes, exit immediately
+	l := &clusterctlv1.ProviderList{}
+	if err := c.List(ctx, l); err == nil {
+		return nil
+	}
```

Comment:
Do we need to check the err != nil condition?

Reply:
done!